vllm/engine/llm_engine.py

```diff
@@ -573,6 +594,11 @@ def step(self) -> List[RequestOutput]:
         if scheduler_outputs.is_empty():
             return ignored
 
+        # only enable speculative decoding for generation run
+        if self.spec_dec_worker and (not scheduler_outputs.prompt_run):
```

Comment:
report a bug here, when we start vllm api server with `python3 -m vllm.entrypoints.api_server --model=/path/to/tgt_model/ --draft-model=/path/to/draft/model/ --propose-cnt=5`, the server errors out. looks like you forgot to `set_draft_tokens` for `step_async` in `AsyncLLMEngine`